common/src/main/java/com/linecorp/centraldogma/common/DefaultChange.java

```diff
@@ -106,6 +110,7 @@
 
         this.path = path;
         this.content = content;
+        this.contentAsText = contentAsText;
```

Comment:
Hmm.. thanks for pointing this out. I think we should handle the case that `Change#ofJsonUpsert` is used to add a new JSON5 file(`ChangeType.ADD`) as it relies on `contentAsText()` which isn't serialized into replication log. 

No need to care about `Change#ofJsonUpsert` to modify the file content(`ChangeType.MODIFY`) since it internally converted into `Change.ofTextPatch`.

I think we should have `Change#contentAsText` in the serialized content to make it work (or a simple workaround would be internally converting `Change#ofJsonUpsert` to `Change#ofTextUpsert` like we did for `ChangeType.MODIFY`).

Reply:
> I wonder if it's difficult to tweak behavior so that serde happens in the same way as we've been doing so far

The type of `content` in `Change#ofJsonUpsert` is `JsonNode` because of the decision we've made for JSON5 entry type (use `JsonNode` over `String`).

Maybe we can do the tweak that `Change` to a JSON5 file have `TextNode` as its `content` and do `if-else` in all places where change gets applied, but this also seems to me extremely unmanageable.

I agree that we should avoid unnecessary complexity, but either way(tweaking serde vs tweak `Change) is not very appealing 😅..

The 3rd(and the most simple) option would be converting `Change#ofJsonUpsert` request to `Change#ofTextUpsert` internally like we already did for `ChangeType.MODIFY`.

Reply:
It turns out that the 3rd option will never work since the serializing and sending replication log happens earlier than we can do conversion.

Reply:
Had a chat with @minwoox and @jrhee17 about this and I was advised to put denormalized JSON5 `Change` to replication logs to preserve its original content.

I tried this approach only to find there's no easy way to pass the original JSON5 content to replication log even with this. `NormalizingPushCommand#changes` gives the original JSON5 content in it, but once it gets JSON-serialized for replication log it loses its JSON5 content since `contentAsText` is ignored in the serialization process.

Meanwhile, it turned out that 3rd option that I mentioned above works just fine.
It works by normalizing JSON5 `Change#ofJsonUpsert` to `Change#ofTextUpsert` and then putting the normalized change to replication log (at ).